### PR TITLE
Add 2 filters to 2 wp_required_field_*-fuctions.

### PR DESCRIPTION
**Pull request by kebbet**
Introduce 2 filters to the 2 new wp_required_field_*-functions.
https://core.trac.wordpress.org/ticket/56389

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-includes/general-template.php [hunk 1]
```diff
@@ -5030,9 +5030,28 @@ function __checked_selected_helper( $helper, $current, $echo, $type ) { // phpcs
...
+	$glyph = __( '*' );
+
+	/**
+	 * Filters the visual indicator for required form fields.
+	 *
+	 * @since 6.1.0
+	 *
+	 * @param string $glyph Character to identify required form fields.
+	 */
+	$glyph = apply_filters( 'wp_required_field_glyph', $glyph );
```
Comment (audrasjb) [inline src/wp-includes/general-template.php]: Let's just provide one filter for the whole HTML content :)

File: src/wp-includes/general-template.php [hunk 2]
```diff
@@ -5030,9 +5030,28 @@ function __checked_selected_helper( $helper, $current, $echo, $type ) { // phpcs
...
 	$indicator = '<span class="required" aria-hidden="true">' . esc_html( $glyph ) . '</span>';
 
+	/**
+	 * Filters the markup for visual indicator for required form fields.
+	 *
+	 * @since 6.1.0
+	 *
+	 * @param string $indicator Markup for the indicator element.
+	 */
+	$indicator = apply_filters( 'wp_required_field_indictor', $indicator );
```
Comment (audrasjb) [inline src/wp-includes/general-template.php]: there's a small typo here ;-)
Replace `wp_required_field_indictor` with `wp_required_field_indicator`

**Review by audrasjb — APPROVED** [review]
Looks good to me, thanks for addressing the two requests :)

File: src/wp-includes/general-template.php [hunk 3]
```diff
@@ -5050,6 +5059,15 @@ function wp_required_field_message() {
...
+	/**
+	 * Filters the message to explain required form fields.
+	 *
+	 * @since 6.1.0
+	 *
+	 * @param string $message Message text and glyph wrapped in a `span` tag.
+	 */
+	$message = apply_filters( 'wp_required_field_message', $message );
+
 	return $message;
```
Comment (costdev) [inline src/wp-includes/general-template.php]:
```suggestion
	return apply_filters( 'wp_required_field_message', $message );
```
This filter seems like it should always be the final value. What do you think @kebbet and @audrasjb? (Same question for both filters)

Comment (kebbet) [reply src/wp-includes/general-template.php]: I have nothing against those changes, can update the PR. Fewer lines of code is indeed cleener.

**Comment by costdev** [comment]
Should we have unit tests to ensure these filters are always fired? @audrasjb

**Comment by audrasjb** [comment]
Ah yes 👍

File: tests/phpunit/tests/general/wpRequiredFieldIndicator.php [hunk 4]
```diff
@@ -0,0 +1,47 @@
+<?php
+/**
+ * Test wp_required_field_indicator().
+ *
+ * @ticket 56389
+ *
```
Comment (costdev) [inline tests/phpunit/tests/general/wpRequiredFieldIndicator.php]:
```suggestion
```

Oops! I shouldn't have had the `@ticket` annotation on the test class itself.

**Review by costdev — CHANGES_REQUESTED** [review]

**Review by mukeshpanchal27 — APPROVED** [review]
The PR looks solid from my side. Great work @costdev and @kebbet

**Review by costdev — APPROVED** [review]
LGTM :+1:

**Comment by costdev** [comment]
@SergeyBiryukov In [80ffb13](https://github.com/WordPress/wordpress-develop/pull/3088/commits/80ffb13dd6391e8b6625801b1031f6c38c2cb070), the test class name was changed from TitleCase to camelCase. In [Writing PHPUnit Tests - Test Classes](https://make.wordpress.org/core/handbook/testing/automated-testing/writing-phpunit-tests/#test-classes), it says:

> Test class names should reflect the filepath, with underscores replacing directory separators and TitleCase replacing camelCase. Thus, the class Tests_Comment_GetCommentClass(), which contains tests for the get_comment_class() function, is located in tests/comment/getCommentClass.php.

I understand this commit was made for consistency. Should the handbook, or the test suite, be updated?

**Review by robinwpdeveloper — APPROVED** [review]

**Comment by audrasjb** [comment]
Committed in https://core.trac.wordpress.org/changeset/54137

**Pull request closed without merging**